src/main/java/org/opensearch/knn/index/vectorvalues/KNNBinaryVectorValues.java

```diff
@@ -34,7 +34,8 @@ public byte[] getVector() throws IOException {
     @Override
     public byte[] conditionalCloneVector() throws IOException {
         byte[] vector = getVector();
-        if (vectorValuesIterator.getDocIdSetIterator() instanceof ByteVectorValues) {
+        if (vectorValuesIterator instanceof KNNVectorValuesIterator.MergeByteVectorValuesIterator
```

Comment:
nit: we could add a wrapper method to unwrap so that we dont always have to check both

Reply:
Its trivial currently so will prefer keeping it as it is, if it gets complicated we can either make a predicate or stub it in a method